prov/shm/src/smr_atomic.c

```diff
@@ -64,15 +66,17 @@ static void smr_format_inline_atomic(struct smr_cmd *cmd,
 	switch (cmd->msg.hdr.op) {
 	case ofi_op_atomic:
 	case ofi_op_atomic_fetch:
-		cmd->msg.hdr.size = ofi_copy_from_iov(cmd->msg.data.msg,
-						SMR_MSG_DATA_LEN, iov, count, 0);
+		cmd->msg.hdr.size = ofi_copy_from_hmem_iov(cmd->msg.data.msg,
```

Comment:
It is possible that ofi_copy_from_hmem_iov() returns an error.

Reply:
How likely is this?  None of the code is structured to handle memcpy suddenly failing.  Can we just assert that hmem memcpy routines will succeed?  Otherwise, the changes that are needed are likely to be far more pervasive.

Reply:
I'm not sure how likely it is, but I agree that asserting success would allow for the changes to be less invasive. I'm fine with asserting it.

Reply:
The only times I have seen a cudaMemcpy fail is if it was passed bad arguments. I think an assert should work.